File: pset1/credit.c
#include <cs50.h>
#include <stdio.h>

/*program that determines whether a provided credit card number 
is valid according to Luhn’s algorithm*/

int main(void)
{
    long long cc;
    long long i; 
    long long remain = 1;
    long long remain_multi = 10;
    int digit;
    int digit_multi = 0;
    int contral_sum = 0;
    
    
    printf("Number:");
    cc = get_long_long();

    for (i = 10 ; i < 10e16; i = i * 100) 
    {
        digit = (cc % i) / remain;
        digit_multi = ((cc % (i * 10)) / remain_multi) * 2;
        remain = remain * 100;
        remain_multi = remain_multi * 100;
        contral_sum = contral_sum + digit;
        
        if (digit_multi > 9) 
        {
            digit_multi = digit_multi % 10 + 1;
        }    
            
        contral_sum = contral_sum + digit_multi; 
    }
    
    if (contral_sum % 10 != 0 )
    {
        printf("INVALID\n");
    }
    
    else if (cc > 10e13 && cc < 10e14) 
    {
    
        int start34 = cc / 10e12;
        int start37 = cc / 10e12;
        
        if (start34 == 34 || start37 == 37)
        {
            printf ("AMEX\n");
        }    
    }    
        
        else if (cc > 10e14 && cc < 10e15) 
        {
    
            int start51 = cc / 10e13;
            int start52 = cc / 10e13;
            int start53 = cc / 10e13;
            int start54 = cc / 10e13;
            int start55 = cc / 10e13;
            int start4_2 = cc / 10e14;
        
            if (start51 == 51 || start52 == 52 || start53 == 53 || start54 == 54 || start55 == 55)
            {
                printf ("MASTERCARD\n");
            }
            
            if (start4_2 == 4)
            {   
                printf ("VISA\n"); 
            }    
        }
     
            else if (cc > 10e11 && cc < 10e12) 
            {
    
                int start4 = cc / 10e11;
        
                if (start4 == 4)
                {    
                    printf ("VISA\n");
                }    
            }
        
            else 
            {
                printf("INVALID\n");
            }
            
}

File: pset6/credit/credit.py
#program that determines whether a provided credit card number
#is valid according to Luhn’s algorithm


def main():

    cc = input("Number: ")
    ccl = list(cc)
    ccl.reverse()
    contral_sum = 0

    for i in range(0, len(ccl)):
        if  (i % 2) != 0:
            digit = int(ccl[i]) * 2
            if digit > 9:
                digit = digit % 10 + 1
                contral_sum = contral_sum + digit
            else:
                contral_sum = contral_sum + digit
        else:
            digit = int(ccl[i])
            contral_sum = contral_sum + digit
    if contral_sum % 10 != 0:
        print("INVALID")

    elif len(ccl) == 15:
        if int(ccl[14]) == 3 and int(ccl[13]) == 7:
            print("AMEX")
    elif len(ccl) == 16:
        if int(ccl[15]) == 5 and int(ccl[14]) == 1 or int(ccl[14]) == 2 or int(ccl[14]) == 3 or int(ccl[14]) == 4 or int(ccl[14]) == 5:
            print("MASTERCARD")
        if int(ccl[15]) == 4:
            print ("VISA")
    elif len(ccl) == 13:
        if int(ccl[12]) == 4:
            print("VISA")
    else:
        print("INVALID")

if __name__ == "__main__":
    main()